crates/cli/src/subcommands/generate/typescript.rs

```diff
@@ -40,16 +40,32 @@ fn scalar_or_string_to_ts(ty: &AlgebraicType) -> Option<(&str, &str)> {
 fn ty_fmt<'a>(ctx: &'a GenCtx, ty: &'a AlgebraicType, ref_prefix: &'a str) -> impl fmt::Display + 'a {
     fmt_fn(move |f| match ty {
         AlgebraicType::Sum(sum_type) => {
-            if let Some(inner_ty) = sum_type.as_option() {
-                write!(f, "{} | null", ty_fmt(ctx, inner_ty, ref_prefix))
+            if sum_type.is_special() {
+                if let Some(inner_ty) = sum_type.as_option() {
+                    write!(f, "{} | null", ty_fmt(ctx, inner_ty, ref_prefix))
+                } else if sum_type.is_schedule_at() {
+                    write!(f, "ScheduleAt")
+                } else {
+                    unimplemented!("Unknown special sum type: {sum_type:?}")
+                }
             } else {
-                unimplemented!()
+                panic!("Cannot format non-special sum type {sum_type:?}")
             }
         }
-        ty if ty.is_identity() => write!(f, "Identity"),
-        ty if ty.is_address() => write!(f, "Address"),
         // All other types should fail.
-        AlgebraicType::Product(_) => unimplemented!(),
+        AlgebraicType::Product(product_type) => {
+            if product_type.is_special() {
```

Comment:
It is not redundant in the presence of code changes. If you add a new `special` type, and miss updating code generation, the old code won't catch it.

I'll revert in favor of doing the enum refactoring later, since that is really the correct way to express this.